static/app/components/dynamicSampling/investigationRule.tsx

```diff
@@ -0,0 +1,227 @@
+      );
+    },
+    onError: (_error: Error) => {
+      addErrorMessage(t('Unable to create investigation rule'));
+    },
+  });
+  return mutate;
+}
+
+const InvestigationInProgressNotification = styled('span')`
+  margin: ${space(1.5)};
+  font-size: ${p => p.theme.fontSizeMedium};
+  color: ${p => p.theme.subText};
+  font-weight: 600;
+  display: inline-flex;
+  align-items: center;
+  gap: ${space(0.5)};
+`;
+
+function InvestigationRuleCreationInternal(props: Props) {
+  const projects = [...props.eventView.project];
+  const organization = props.organization;
+  const period = props.eventView.statsPeriod || null;
+  const query = props.eventView.getQuery();
+  const createInvestigationRule = useCreateInvestigationRuleMutation({
+    query,
+    projects,
+    organization,
+    period,
+  });
+  const request = useGetExistingRule(query, projects, organization, props.numSamples);
+
+  if (!hasTooFewSamples(props.numSamples)) {
+    // no results yet (we can't take a decision) or enough results,
+    // we don't need investigation rule UI
+    return null;
+  }
+  if (request.isLoading) {
+    return null;
+  }
+
+  if (request.error !== null) {
+    const errorResponse = t('Unable to fetch investigation rule');
+    addErrorMessage(errorResponse);
+    return null;
+  }
+
+  const rule = request.data;
+  const haveInvestigationRuleInProgress = rule !== null;
+
+  if (haveInvestigationRuleInProgress) {
+    // investigation rule in progress, just show a message
+    const existingRule = rule as CustomDynamicSamplingRule;
+    const ruleStartDate = new Date(existingRule.startDate);
+    const now = new Date();
+    const interval = moment.duration(now.getTime() - ruleStartDate.getTime()).humanize();
+
+    return (
+      <InvestigationInProgressNotification>
+        {tct('Collecting samples since [interval]  ago.', {interval})}
+
+        <Tooltip
+          isHoverable
+          title={tct(
+            'A user has temporarily adjusted retention priorities, increasing the odds of getting events matching your search query. [link:Learn more.]',
+            // TODO find out where this link is pointing to
+            {
+              link: <ExternalLink href="https://docs.sentry.io" />,
+            }
+          )}
+        >
+          <IconQuestion size="xs" color="subText" />
+        </Tooltip>
+      </InvestigationInProgressNotification>
+    );
+  }
+
+  // no investigation rule in progress, show a button to create one
+  return (
+    <Tooltip
+      isHoverable
+      title={tct(
+        'We can find more events that match your search query by adjusting your retention priorities for an hour, increasing the odds of getting matching events. [link:Learn more.]',
+        // TODO find out where this link is pointing to
+        {
+          link: <ExternalLink href="https://docs.sentry.io" />,
+        }
+      )}
+    >
+      <Button
+        size="sm"
```

Comment:
Discover is using "sm", performance needs "xs". Let's add a `buttonProps` prop to the InvestigationRule component that will  explode them inside this button

Reply:
Done